src/pipelinedefinition.cpp

```diff
@@ -1320,16 +1320,16 @@ shape_t PipelineDefinition::getNodeGatherShape(const NodeInfo& info) const {
                     if (nodeOutputsInfo.size() == 0) {
                         SPDLOG_ERROR("Node: {} library metadata reports no outputs", nodeName);
                         return;
-                    } else if (nodeOutputsInfo.begin()->second->getShape().size() < 3) {
+                    } else if (nodeOutputsInfo.begin()->second->getShape_3().getFlatShape().size() < 3) {
```

Comment:
Why do we really need to call `getFlatShape` here? This always has the same value as `getShape_3().getSize()`.

Reply:
true